test/acl.test.js

```diff
@@ -472,3 +476,130 @@ describe('access check', function() {
       });
   });
 });
+
+describe('Authorized roles propagated in RemotingContext', function() {
+  var app, request, accessToken;
+  var models = {};
+
+  beforeEach(setupAppAndRequest);
+
+  it('contains all authorized roles for a principal if query is allowed', function() {
+    return createACLs('MyTestModel', [
+      {permission: ACL.ALLOW, principalId: '$everyone'},
+      {permission: ACL.ALLOW, principalId: '$authenticated'},
+      {permission: ACL.ALLOW, principalId: 'myRole'},
+    ])
+    .then(function() {
+      return request.get('/MyTestModels')
+        .set('X-Access-Token', accessToken.id)
+        .expect(200);
+    })
+    .then(function() {
+      var ctx = models.MyTestModel.lastRemotingContext;
+      expect(ctx.args.options.authorizedRoles).to.eql(
+        {
+          $everyone: true,
+          $authenticated: true,
+          myRole: true,
+        }
+      );
+    });
+  });
+
+  it('does not contain any denied role even if query is allowed', function() {
+    return createACLs('MyTestModel', [
+      {permission: ACL.ALLOW, principalId: '$everyone'},
+      {permission: ACL.DENY,  principalId: '$authenticated'},
+      {permission: ACL.ALLOW, principalId: 'myRole'},
+    ])
+    .then(function() {
+      return request.get('/MyTestModels')
+        .set('X-Access-Token', accessToken.id)
+        .expect(200);
+    })
+    .then(function() {
+      var ctx = models.MyTestModel.lastRemotingContext;
+      expect(ctx.args.options.authorizedRoles).to.eql(
+        {
+          $everyone: true,
+          myRole: true,
+        }
+      );
+    });
+  });
+
+  it('honors default permission setting', function() {
+    // default permission is set to DENY for MyTestModel
+    models.MyTestModel.settings.defaultPermission = ACL.DENY;
+
+    return createACLs('MyTestModel', [
+      {permission: ACL.DEFAULT, principalId: '$everyone'},
+      {permission: ACL.DENY,    principalId: '$authenticated'},
+      {permission: ACL.ALLOW,   principalId: 'myRole'},
+    ])
+    .then(function() {
+      return request.get('/MyTestModels')
+        .set('X-Access-Token', accessToken.id)
+        .expect(200);
+    })
+    .then(function() {
+      var ctx = models.MyTestModel.lastRemotingContext;
+      expect(ctx.args.options.authorizedRoles).to.eql(
+        // '$everyone' is not expected as default permission is DENY
+        {myRole: true}
+      );
+    });
+  });
+
+  // helpers
+  function setupAppAndRequest() {
+    app = loopback({localRegistry: true, loadBuiltinModels: true});
+    app.use(loopback.rest());
+    app.set('remoting', {errorHandler: {debug: true, log: true}});
+    app.enableAuth();
+    app.dataSource('db', {connector: 'memory'});
+    request = supertest(app);
+
+    app.enableAuth({dataSource: 'db'});
+    models = app.models;
+
+    // creating a custom model
+    models.MyTestModel = app.registry.createModel('MyTestModel');
+    app.model(models.MyTestModel, {dataSource: 'db'});
```

Comment:
Uff, modifying `app.models` directly is not a good practice. I am proposing the following:

```js
const MyTestModel = app.registry.createModel('MyTestModel');
app.model(MyTestModel, {dataSource: 'db'});
// models.MyTestModel was set by app.model() by now
```